spi/src/main/java/com/flipkart/varadhi/spi/services/PolledMessage.java

```diff
@@ -0,0 +1,23 @@
+package com.flipkart.varadhi.spi.services;
+
+import com.flipkart.varadhi.entities.Offset;
+
+/**
+ * Represents a single message polled from some partition.
+ */
+// TODO: Keeping payload as byte[] only for now. When implementing, there will be opportunity to evaluate how to avoid
+// unnecessary deserialization / array copy & then we can tweak this interface.
+public interface PolledMessage<O extends Offset> {
+
+    /**
+     * @return the offset of this message in the partition.
+     */
+    O offset();
+
+    /**
+     * @return the payload of this message.
+     */
+    byte[] payload();
```

Comment:
what about headers ? should it be message object instead with those semantics attached ?

Reply:
I dont remember, but did we have some discussion around this before?

Reply:
should I just refer entities.Message here instead of this payload array?